modules/Staff/staff_manage_edit.php

```diff
@@ -157,140 +155,72 @@
-                    echo '</div>';
+                    $contractsGateway = $container->get(StaffContractGateway::class);
 
-                    if ($result->rowCount() < 1) {
-                        echo "<div class='error'>";
-                        echo __('There are no records to display.');
-                        echo '</div>';
-                    } else {
-                        echo "<table cellspacing='0' style='width: 100%'>";
-                        echo "<tr class='head'>";
-                        echo '<th>';
-                        echo __('Title');
-                        echo '</th>';
-                        echo '<th>';
-                        echo __('Status').'<br/>';
-                        echo '</th>';
-                        echo '<th>';
-                        echo __('Dates');
-                        echo '</th>';
-                        echo '<th>';
-                        echo __('Actions');
-                        echo '</th>';
-                        echo '</tr>';
-
-                        $count = 0;
-                        $rowNum = 'odd';
-                        while ($row = $result->fetch()) {
-                            if ($count % 2 == 0) {
-                                $rowNum = 'even';
-                            } else {
-                                $rowNum = 'odd';
-                            }
-                            ++$count;
-
-                            echo "<tr class=$rowNum>";
-                            echo '<td>';
-                            echo $row['title'];
-                            echo '</td>';
-                            echo '<td>';
-                            echo $row['status'];
-                            echo '</td>';
-                            echo '<td>';
-                            if ($row['dateEnd'] == '') {
-                                echo dateConvertBack($guid, $row['dateStart']);
+                    $criteria = $contractsGateway->newQueryCriteria()
+                        ->sortBy(["dateStart"], 'DESC');
+
+                    $contracts = $contractsGateway->queryContractsByStaff($criteria, $gibbonStaffID);
+
+                    $table = DataTable::create('contracts');
+
+                    $table->addHeaderAction('add', __('Add'))
+                        ->setURL('/modules/Staff/staff_manage_edit_contract_add.php')
+                        ->addParam('gibbonStaffID', $gibbonStaffID)
+                        ->addParam('search', $search);
+
+                    $table->addColumn('title', __('Title'));
+                    $table->addColumn('status', __('Status'));
+                    $table->addColumn('dates', __('Dates'))
+                        ->format(function ($row) use ($guid) {
+                            if ($row["dateEnd"] == '') {
+                                return dateConvertBack($guid, $row['dateStart']);
```

Comment:
dateConvertBack is deprecated, there's a Format::date function to replace it, which has the same parameters minus the $guid. In this case there's also a Format::dateRange method which takes a start and end and automatically handles the conditional for checking if they're the same date :)